compiler/src/dotty/tools/dotc/typer/QuotesAndSplices.scala

```diff
@@ -119,14 +119,34 @@ trait QuotesAndSplices {
             EmptyTree
         }
       }
+      val typedTypeargs = tree.typeargs.map {
+        case typearg: untpd.Ident =>
+          val typedTypearg = typedType(typearg)
+          /* TODO-18271: Allow type bounds?
+           * (NOTE: Needs non-trivial extension to type system)
+           */
+          val bounds = ctx.gadt.fullBounds(typedTypearg.symbol)
+          if bounds != null && bounds != TypeBounds.empty then
```

Comment:
We might want to move this logic to `checkPattern` because we have fully-typed tree.